File: almostIncreasingSequence/almostIncreasingSequence.ts
function almostIncreasingSequence(sequence: number[]): boolean {

	const decreases = sequence.reduce( (gatherer, curr, currIdx) => (
			sequence[currIdx+1] < curr 
				? [ ...gatherer, curr ]
				: gatherer )
		, []);

	return decreases.length <= 1;
}

console.time('almostIncreasingSequence x2')

console.log(almostIncreasingSequence([1, 3, 2, 1])) 
console.log(almostIncreasingSequence([1, 3, 2])) 

console.timeEnd('almostIncreasingSequence x2')

File: almostIncreasingSequence/almost_increasing_sequence.py

from typing import List


def almost_increasing_sequence(sequence: List[int]) -> bool:
    return (len([n for i, n in enumerate(sequence)
            if 0 <= (i+1) < len(sequence) and
            sequence[i+1] < n]) <= 1)


print(almost_increasing_sequence([1, 3, 2, 1]))
print(almost_increasing_sequence([1, 3, 2]))
